packages/block-library/src/file/editor.scss

```diff
@@ -29,16 +23,27 @@
 		left: 0;
 	}
 
+	// This is an extra container present only in the editor view.
 	.wp-block-file__content-wrapper {
-		flex-grow: 1;
+		display: flex;
+		flex: 1;
+		align-items: center;
 	}
 
+	// This is the editable filename on the left.
+	// Make sure it matches the styling on the frontend.
+	// Except for flex-grow, it is useful to have an easily selectable block.
 	a {
-		min-width: 1em;
+		min-width: 2em !important;
 	}
 
+	// This element maps to the Download button link on the right.
 	.wp-block-file__button-richtext-wrapper {
-		display: inline-block;
-		margin-left: 0.75em;
+		margin-left: auto;
+	}
+
+	// The file block already has a border, so don't show the extra border of the placceholder.
+	.components-placeholder.components-placeholder {
+		box-shadow: none;
```

Comment:
I suppose if we _do_ add border support, you could set the border to zero which might be weird in the placeholder state 🤔 

Then again, it seems like the placeholder state itself could do with a great deal of iteration to be more minimal — almost just a small dashed-line dropzone of the same size as the resulting block? Probably best to explore separately.